Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -589,18 +585,73 @@ PlatformImagePtr ImageDecoderCG::createFrameImageAtIndex(size_t index, Subsampli
     return maskedImage ? maskedImage : image;
 }
 
+String ImageDecoderCG::decodeUTI(const SharedBuffer& data) const
+{
+    auto uti = String(CGImageSourceGetType(m_nativeDecoder.get()));
+    if (uti != "public.heif"_s && uti != "public.heic"_s)
+        return uti;
+    
+    constexpr auto bytesToUnsigned = [](const char bytes[4]) -> unsigned {
+        return bytes[0] | bytes[1] << 8 | bytes[2] << 16 | bytes[3] << 24;
+    };
+
+    static constexpr unsigned ftypSignature = bytesToUnsigned("ftyp");
+    static constexpr unsigned avifBrand = bytesToUnsigned("avif");
+    static constexpr unsigned avisBrand = bytesToUnsigned("avis");
+
+    // HEIF/AVIF files start with an ftyp box
+    //
+    //  aligned(8) class GeneralTypeBox(code) extends Box(code) {
+    //      unsigned int(32)    major_brand;
+    //      unsigned int(32)    minor_version;
+    //      unsigned int(32)    compatible_brands[];    // to end of the box
+    //  }
+    //
+    // A box starts with 4 bytes for the size (including the length bytes).
+    // An image is considered an AVIF if "avif" or "avis" is present in its
+    // major_brand or in one of the compatible_brands.
+
+    const unsigned* ftypBox = reinterpret_cast<const unsigned*>(data.data());
```

Comment:
This code will be executed when` CGImageSourceGetType()` returns `public.heif`. And this can't happen if the `data.size()` is less than 4. So I added an ASSERT_NOT_REACHED().